### PR TITLE
Combine OCS API getUser method code into provisioning_api app

### LINKED ISSUE #13002: Combine OCS API getUser method code into provisioning_api app.
Currently we have two code sections for the same api method, and they are merged:

https://github.com/owncloud/core/blob/master/lib/private/ocs/cloud.php#L63 and
https://github.com/owncloud/core/pull/13000/files#diff-d63ef796ade5860b68adcc00dc5639a6R68.

We should move all of the code into the provisioning_api app to cleanup.

### PR DESCRIPTION
**Pull request by rullzer**
Fixes #13002

Move the cloud/users/{userid} code in total to the provisioning API.

### CODE REVIEW

**Comment by rullzer** [comment]
CC: @tomneedham

File: apps/provisioning_api/lib/users.php [hunk 1]
```diff
@@ -115,46 +115,28 @@ public function getUser($parameters){
...
-				'email',
-				'displayname',
-				);
 		}
 
 		// Find the data
-		$data = [];
-		$data = self::fillStorageInfo($userId, $data);
-		$data['enabled'] = $this->config->getUserValue($userId, 'core', 'enabled', 'true');
+		$data['quota'] = self::fillStorageInfo($userId, $data);
```
Comment (MorrisJobke) [inline apps/provisioning_api/lib/users.php]: Why are you still pass in the `$data` parameter?

Comment (rullzer) [inline apps/provisioning_api/lib/users.php]: I wanted to make sure people are actually reviewing my PR's ;)

Nah, just forgot. Fixed

File: ocs/routes.php [hunk 2]
```diff
@@ -91,13 +91,6 @@
 	);
 API::register(
 	'get',
-	'/cloud/users/{userid}',
```
Comment (PVince81) [inline ocs/routes.php]: this looks suspicious, removing this endpoint?

Comment (tomneedham) [inline ocs/routes.php]: Yes both core and provisioning_api apps used to provide results to this endpoint so there is no longer the need to register the route in core.

Comment (tomneedham) [inline ocs/routes.php]: https://github.com/owncloud/core/blob/master/apps/provisioning_api/appinfo/routes.php#L38

Comment (MorrisJobke) [inline ocs/routes.php]: This combining all results of one route will drive me crazy :see_no_evil: :hear_no_evil: :speak_no_evil:

Comment (tomneedham) [inline ocs/routes.php]: @MorrisJobke But the PR is to begin removing this....

**Comment by PVince81** [comment]
@SergioBertolinSG can you run your magical tests against this branch ?

**Comment by DeepDiver1975** [comment]
@PVince81 

They pass :wink: 

bildschirmfoto von 2015-10-20 11-09-30

**Comment by DeepDiver1975** [comment]
:+1:

**Pull request merged**